Signal/src/AppDelegate.m

```diff
@@ -505,6 +461,56 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
     // Always check prekeys after app launches, and sometimes check on app activation.
     [TSPreKeyManager checkPreKeysIfNecessary];
 
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        // At this point, potentially lengthy DB locking migrations could be running.
+        // Avoid blocking app launch by putting all further possible DB access in async thread.
+        [[TSAccountManager sharedInstance]
+            ifRegistered:YES
+                runAsync:^{
+                    DDLogInfo(@"%@ running post launch block for registered user: %@",
+                        self.tag,
+                        [TSAccountManager localNumber]);
+
+                    [TSSocketManager requestSocketOpen];
+
+                    RTCInitializeSSL();
+
+                    [OWSSyncPushTokensJob runWithPushManager:[PushManager sharedManager]
+                                              accountManager:[Environment getCurrent].accountManager
+                                                 preferences:[Environment preferences]
+                                                  showAlerts:NO];
+
+                    // Clean up any messages that expired since last launch immediately
+                    // and continue cleaning in the background.
+                    [[OWSDisappearingMessagesJob sharedJob] startIfNecessary];
+
+                    // Mark all "attempting out" messages as "unsent", i.e. any messages that were not successfully
+                    // sent before the app exited should be marked as failures.
+                    [[[OWSFailedMessagesJob alloc] initWithStorageManager:[TSStorageManager sharedManager]] run];
+                    [[[OWSFailedAttachmentDownloadsJob alloc] initWithStorageManager:[TSStorageManager sharedManager]]
+                        run];
+
+                    [AppStoreRating setupRatingLibrary];
+                }];
+
+        [[TSAccountManager sharedInstance]
+            ifRegistered:NO
+                runAsync:^{
+                    dispatch_async(dispatch_get_main_queue(), ^{
+                        DDLogInfo(@"%@ running post launch block for unregistered user.", self.tag);
+                        [TSSocketManager requestSocketOpen];
```

Comment:
Huh, I realize you just copy/pasted this, but it seems weird to be trying to open the websocket when we're not registered. Maybe it's just to get the retry-timer going for when we complete registration.